src/get-started/java-binding.md

```diff
@@ -1371,9 +1371,47 @@ service sources and place it inside some known directory.
 Finally, add an `instances` section that describes the list of service instances
 you want to start in the blockchain. For each instance specify its
 artifact name alias in the `artifact` field. Instance names are the keys in the
-YAML dictionary. The Instance Configuration plugin also requires a list of
-additional parameters. In the present example the parameters refer to the
-`timestamping` instance:
+YAML dictionary.
+
+```yaml
+instances:
+  xnm-token:
+    artifact: cryptocurrency
+  time-oracle:
+    artifact: time
+```
+
+To instantiate a service which requires configuration parameters,
+`config` dictionary must be supplied.
+
+The Instance Configuration plugin supports several configuration formats:
+
+- Standard text formats: text, JSON, [java.util.Properties].
+- Arbitrary Protocol Buffers messages.
+
+#### Text Configuration Formats
+
+Requires the following parameters:
+
+- `format`. Describes the format of the configuration string. Possible values
+  are: `text`, `json` and `properties` (see [java.util.Properties]).
+- `value`. A configuration string, may be used instead of `from_file`. If both
+  `value` and `from_file` are present, `value` takes higher priority.
+- `from_file`. A path to a file containing configuration string. May be absolute
+  or relative to the working directory.
+
+```yaml
+instances:
+  timestamping:
+    artifact: timestamping
+    config:
+      format: "properties"
+      value: "time_service_name=time"
```

Comment:
it'll be great to see in the example how to specify a few pairs 🙃

Reply:
> how

Certainly not inline :-)

Reply:
By using `.properties` file, yes

Reply:
great then, our doc looks like a usual tutorial :-)

https://www.youtube.com/watch?v=MAlSjtxy5ak

Reply:
Well, I can replace `properties` by `text` in this example. Would it work?

Reply:
But do we need to document how to specify a string path to a file, or to write _service-specific_ configuration (in JSON, properties, plain text)?

If `value` is useful only for a single arguments (I have no idea how to specify multi-line string in TOML), we may use value_from_file as an example instead.